pkg/reconciler/labeler/v1/labels.go

```diff
@@ -127,15 +124,15 @@ func setLabelForListed(route *v1.Route, acc accessor, names sets.String) error {
 // deleteLabelForNotListed uses the accessor to delete the label from any listable entity that is
 // not named within our list.  Unlike setLabelForListed, this function takes ns/name instead of a
 // Route so that it can clean things up when a Route ceases to exist.
-func deleteLabelForNotListed(ctx context.Context, ns, name string, acc accessor, names sets.String) error {
+func deleteLabelForNotListed(ns, name string, acc Accessor, names sets.String) error {
 	oldList, err := acc.list(ns, name)
 	if err != nil {
 		return err
 	}
 
 	// Delete label for newly removed traffic targets.
 	for _, elt := range oldList {
-		if names.Has(elt.GetName()) {
+		if names != nil && names.Has(elt.GetName()) {
```

Comment:
no need to check `names` for nil. Map readonly operations are safe on nil maps.

Reply:
oh neat